gradle/build-scan.gradle

```diff
@@ -19,12 +26,22 @@ buildScan {
         buildScan.server = 'https://gradle-enterprise.elastic.co'
     }
 
+    if (nodeName) {
+        link 'System logs', "https://infra-stats.elastic.co/app/infra#/logs?" +
+                "&logFilter=(expression:'host.name:${nodeName}',kind:kuery)"
+        link 'System metrics', "https://infra-stats.elastic.co/app/infra#/metrics/host/" +
+                "${nodeName}?_g=()&metricTime=(autoReload:!f,refreshInterval:5000," +
+                "time:(from:${startTime -  TimeUnit.MILLISECONDS.convert(5, TimeUnit.MINUTES)},interval:%3E%3D1m," +
+                "to:${System.currentTimeMillis() +  TimeUnit.MILLISECONDS.convert(5, TimeUnit.MINUTES)}))"
```

Comment:
I'm curious how this even works. I guess because this is a `GString` it's evaulation must be deferred somehow? Wouldn't `System.currentTimeMillies()` here evaluate to the time this script is evaluated, which is when the build is started?

We should use [`buildScan.buildFinished { }`](https://docs.gradle.com/enterprise/gradle-plugin/api/com/gradle/scan/plugin/BuildScanExtension.html#buildFinished(org.gradle.api.Action)) here instead to ensure we generate this link at the end of the build.

Reply:
Good catch. Indeed this needs to be in `buildFinished` and we can use current time for the start time, I got it all wrong.